### DISCUSSION TITLE
Add a role folder for experimental roles

### DISCUSSION THREAD

**Discussion by oxedions** [bluebanquise/bluebanquise#464] [Ideas]
Hello,

A user is asking for creation of another role folder, called "experimental", that would be dedicated to huge roles that could not be easy to validate, and so require a medium / long term testing before being validated as stable.

Any thought regarding this ?

**Comment by strus38** [comment]
The user is me :-)... This comes from a PR I would like to do on kerberos client/server which is not as simple as installing a few RPMs, and which requires a lot of experience on kerberos itself which is not widespread.

**Reply by btravouillon** [reply]
I'm not sure to understand. Do you want to merge some experimental code in master? Can't you use a feature branch in your fork instead to develop your roles, then once ready, open a PR?

**Reply by strus38** [reply]
the code that I will submit will be tested as far as I can (kerberos can be tricky sometimes). What I would like is separate this role from others which we know are perfectly working since we have a lot of experience on them... in case of Kerberos, the idea would be that we will wait some feedbacks from several deployments before deciding that it is 'good enough'.

**Reply by btravouillon** [reply]
In such use case I would tend to use a separate git project while the role is not considered stable. Any user with an interest in this role can install it under `custom` directory (or create an `experimental` directory and update ansible.cfg accordingly).
Once you get positive feedbacks on your role and you consider it  to be ready for production, you can open a PR with the code.

It may be possible to host experimental roles in the GitHub bluebanquise namespace.

**Comment by strus38** [comment]
Then I fear no one will never use it.

Le ven. 6 nov. 2020 à 14:54, Bruno Travouillon <notifications@github.com> a
écrit :

**Comment by strus38** [comment]
So drop the idea, I will make it ready to be put in addons.

Le ven. 6 nov. 2020 à 14:56, Indigo Ping <indigoping4cgmi@gmail.com> a
écrit :
